<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>703</width>
    <height>547</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Nonogram</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTableView" name="tableView">
      <property name="editTriggers">
       <set>QAbstractItemView::AnyKeyPressed|QAbstractItemView::DoubleClicked|QAbstractItemView::EditKeyPressed|QAbstractItemView::SelectedClicked</set>
      </property>
      <property name="verticalScrollMode">
       <enum>QAbstractItemView::ScrollPerPixel</enum>
      </property>
      <property name="horizontalScrollMode">
       <enum>QAbstractItemView::ScrollPerPixel</enum>
      </property>
      <attribute name="horizontalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderMinimumSectionSize">
       <number>0</number>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderMinimumSectionSize">
       <number>0</number>
      </attribute>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>703</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_new"/>
    <addaction name="action_open"/>
    <addaction name="action_save"/>
    <addaction name="separator"/>
    <addaction name="action_quit"/>
   </widget>
   <widget class="QMenu" name="menuSolve">
    <property name="title">
     <string>Solve</string>
    </property>
    <addaction name="action_solve"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <widget class="QMenu" name="menuRows">
     <property name="title">
      <string>Row</string>
     </property>
     <addaction name="action_row_insert_above"/>
     <addaction name="action_row_insert_below"/>
     <addaction name="action_row_remove_current"/>
    </widget>
    <widget class="QMenu" name="menuColumn">
     <property name="title">
      <string>Column</string>
     </property>
     <addaction name="action_column_insert_left"/>
     <addaction name="action_column_insert_right"/>
     <addaction name="action_column_remove_current"/>
    </widget>
    <addaction name="menuRows"/>
    <addaction name="menuColumn"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menuEdit"/>
   <addaction name="menuSolve"/>
  </widget>
  <action name="action_open">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="action_save">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="action_quit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="action_new">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="action_solve">
   <property name="text">
    <string>Solve</string>
   </property>
  </action>
  <action name="actionAdd_row">
   <property name="text">
    <string>Add row below</string>
   </property>
  </action>
  <action name="actionAdd_column">
   <property name="text">
    <string>Add column right</string>
   </property>
  </action>
  <action name="actionRemove_row">
   <property name="text">
    <string>Remove current row</string>
   </property>
  </action>
  <action name="actionRemove_column">
   <property name="text">
    <string>Remove current column</string>
   </property>
  </action>
  <action name="action_row_insert_above">
   <property name="text">
    <string>Insert above</string>
   </property>
  </action>
  <action name="action_row_insert_below">
   <property name="text">
    <string>Insert below</string>
   </property>
  </action>
  <action name="action_row_remove_current">
   <property name="text">
    <string>Remove current</string>
   </property>
  </action>
  <action name="action_column_insert_left">
   <property name="text">
    <string>Insert left</string>
   </property>
  </action>
  <action name="action_column_insert_right">
   <property name="text">
    <string>Insert right</string>
   </property>
  </action>
  <action name="action_column_remove_current">
   <property name="text">
    <string>Remove current</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>action_new</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileNew()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_open</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_save</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_solve</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>solve()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_row_insert_above</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>editRowInsertAbove()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_row_insert_below</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>editRowInsertBelow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_column_insert_left</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>editColumnInsertLeft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_column_insert_right</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>editColumnInsertRight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_column_remove_current</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>editColumnRemoveCurrent()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_row_remove_current</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>editRowRemoveCurrent()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>fileNew()</slot>
  <slot>fileOpen()</slot>
  <slot>fileSave()</slot>
  <slot>solve()</slot>
  <slot>editRowInsertAbove()</slot>
  <slot>editRowInsertBelow()</slot>
  <slot>editRowRemoveCurrent()</slot>
  <slot>editColumnInsertLeft()</slot>
  <slot>editColumnInsertRight()</slot>
  <slot>editColumnRemoveCurrent()</slot>
 </slots>
</ui>
